### PR TITLE
Organizer can add missions

### LINKED ISSUE #431: Organizers can update details on a mission from mission control
**Description**

As an organizer, I can open and edit full details of a mission from within mission controls so that I can update missions as I organize them and also so that I can deal with unexpected developments as they occur.

[Airtable #162](https://airtable.com/tblRjd2McKgRW9MKj/viwlRRNI0PZIxrGkc/recUd4ZiLI1THmka5?blocks=hide)

[Figma](https://www.figma.com/file/GLg0zPNnwMEGIb9TytCrxQ/Resilience-App?node-id=29260%3A21008)

### LINKED ISSUE #565: Organizers can create new mission and specify mission details (how many of which food box) and then be taken to mission edit view
Part of the mission edit user story - #431

**Organizers may also need to create entirely new food box missions as part of MVP0.  To do this they need to be able to specify which foodboxes are in the mission and how many.** 

That means bringing this type of editing into the 'create mission' for organizers. 

Hopefully, some code can be re-used or copy and pasted from  request/foodbox/FoodBoxStep.jsx 

I don't want to do premature design but I would imagine this will be easier as a step *before* diving into the mission edit view that an organizer sees when they click 'edit' on an existing mission. 

Again apologies for what may be premature design but I imagine that this step creates a 'blank' mission that you can click through to the real mission. However there may be other better options for that, so this is just a strawman idea.

The data we're editing here is "mission details" specifically fields that look like this on a mission.

```json
"type": "resource",
"details": [
  {
    "resourceUid": "1",
    "displayName": "Produce Box #1",
    "quantity": 3,
    "type": "foodbox",
    "cost": "25",
    "description": "Frisee and Head Lettuce box - one of every head of lettuce, 1/2 lb bag baby spinach, 1/4lb Italian parsley, 1/4lb cilantro"
  },
  {
    "resourceUid": "1",
    "displayName": "Produce Box #1",
    "quantity": 3,
    "type": "foodbox",
    "cost": "25",
    "description": "Frisee and Head Lettuce box - one of every head of lettuce, 1/2 lb bag baby spinach, 1/4lb Italian parsley, 1/4lb cilantro"
  }
]
```

### PR DESCRIPTION
**Pull request by thirdeyeclub**
May be a duplicate of #565

### CODE REVIEW

**Comment by jwu910** [comment]
@thirdeyeclub Your description says this is a duplicate. Can we close?

File: src/app/dashboard/Missions/CreateMission.js [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+                return (
+                  <div className={classes.sideBySideSelect} key={boxNumber}>
+                    <Select
+                      onChange={handleItems}
+                      name={"item for box " + boxNumber}
+                      variant="outlined"
+                      className={classes.select}
+                      value={name}
+                    >
+                      <MenuItem value="Fruits and Veggies">Fruits & Veggies</MenuItem>
```
Comment (amylo) [inline src/app/dashboard/Missions/CreateMission.js:259]: @thirdeyeclub  Please populate this dropdown dynamically with the different types of Foodboxes that are offered by the organization.  We can't hardcode the values here.

File: src/app/dashboard/Missions/CreateMission.js [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+            className={classes.select}
+            value={missionType}
+            onChange={handleChangeMissionType}
+          >
+            <MenuItem className={classes.menuItem} value="food-box">
+              Food box
+            </MenuItem>
+          </Select>
+        </FormControl>
+        {missionType !== null ? (
```
Comment (amylo) [inline src/app/dashboard/Missions/CreateMission.js:244]: @thirdeyeclub You don't need to use a ternary if you actually only check for one condition. You can use `&&`. See https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

File: src/app/dashboard/Missions/CreateMission.js [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+            value={missionFund}
+            onChange={handleChangeFund}
+            startAdornment={
+              <InputAdornment position="start">
+                <AttachMoneyIcon style={{ color: "#3739B5" }} />
+              </InputAdornment>
+            }
+          >
+            {/* THIS WILL HAVE TO HAVE A DYNAMIC INITIAL VALUE */}
+            <MenuItem className={classes.menuItem} value="Not yet funded">
```
Comment (amylo) [inline src/app/dashboard/Missions/CreateMission.js:300]: @thirdeyeclub will you be adding dynamic values here?

File: src/app/dashboard/Missions/CreateMission.js [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+              Not yet funded
+            </MenuItem>
+            <MenuItem className={classes.menuItem} value="Funded">
+              Funded
+            </MenuItem>
+          </Select>
+        </FormControl>
+        <Divider variant="middle" />
+        <div className={classes.dualForm}>
+          <FormControl className={classes.form2}>
```
Comment (amylo) [inline src/app/dashboard/Missions/CreateMission.js:310]: @thirdeyeclub Can you name the style classes with something more meaningful than "form2", eg. "recipientDetails"?

File: src/app/dashboard/Missions/CreateMission.js [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+              </MenuItem>
+            </Select>
+            <br />
+            <TextField
+              variant="outlined"
+              name="phoneNumber"
+              onChange={handleRecipient}
+              className={classes.dualText}
+              value={recipient.phoneNumber}
+              labelid="phone"
```
Comment (amylo) [inline src/app/dashboard/Missions/CreateMission.js:335]: @thirdeyeclub This should be `labelId`. See https://material-ui.com/api/select/#main-content

File: src/app/dashboard/Missions/CreateMission.js [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+              placeholder="(123) 456-7890"
+              InputProps={{
+                startAdornment: (
+                  <InputAdornment position="start">
+                    <PhoneIcon style={{ color: "#3739B5" }} />
```
Comment (amylo) [inline src/app/dashboard/Missions/CreateMission.js:340]: @thirdeyeclub  Can you move this inline style out into the styles object? and we should be referring to theme colours. Hardcoding styles here one-off will make it difficult to maintain.

File: src/app/dashboard/Missions/CreateMission.js [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+                ),
+              }}
+            />
+          </FormControl>
+        </div>
+        <Divider variant="middle" />
+        <div className={classes.dualForm}>
+          <FormControl className={classes.form2}>
+            <Typography className={classes.inputHeader2}>Drop Off Details </Typography>
+            <TextField
```
Comment (amylo) [inline src/app/dashboard/Missions/CreateMission.js:387]: @thirdeyeclub There should be an AddressInput you can use

**Review by amylo — COMMENTED** [review]
@thirdeyeclub As a general comment, it would be best to create reusable components or functions when you notice yourself writing the same thing over and over

**Comment by utunga** [comment]
@amylo  thanks so much for the very detailed feedback. I hope this is useful for you to understand why this is not yet merged @thirdeyeclub . 

That said I'm worried about the fact that we also have some very closely related work going on over in the 'edit mission' epic. We should probably have you and @laredotornado  connect on the question of what works and/or components can be reused between the two user stories? 

Probably the best thing would be to let you two (and perhaps others) work out how to attack things from here. Perhaps best discussed on the channel? I just wanted to flag it as a possibility that there is a lot of reuse that can be done here between the two different stories.

File: src/app/dashboard/Missions/CreateMission.js (hunk 1 above)
Comment (thirdeyeclub) [reply src/app/dashboard/Missions/CreateMission.js:259]: When I wrote this they did not exist at all yet.

File: src/app/dashboard/Missions/CreateMission.js (hunk 3 above)
Comment (thirdeyeclub) [reply src/app/dashboard/Missions/CreateMission.js:300]: They did not exist at the time 20+ days ago when I wrote it.

**Comment by thirdeyeclub** [comment]
I mean if it wasn't sitting in PR for almost a month - with someone just looking at it now - this wouldn't be happening. We should have meeting so we can look over all the complements and she how they align with the edit mission epic.

**Comment by jwu910** [comment]
@thirdeyeclub I have to disagree. This project has been moving a mile a minute and given the fluidity of the features and changing demand, features are bound to change.

Every time I pull down master and I see code has changed, I check all my open PRs and see if merged changes has affected the branches I worked off of. If they do, I always rebase on to upstream/master to bring my feature branch up to date. 

We should always have our features assumed to be built off of the latest code.

**Comment by utunga** [comment]
Yayyy 🔥🔥🔥 you did it @jwu910 :-)

**Comment by jwu910** [comment]
> Yayyy 🔥🔥🔥 you did it @jwu910 :-)

Lol hope you're well! Closing up some stale PRs that have been sitting around :(

**Pull request closed without merging**